docs/sources/flow/reference/components/discovery.dockerswarm.md

```diff
@@ -0,0 +1,238 @@
+---
+aliases:
+  - /docs/grafana-cloud/agent/flow/reference/components/discovery.dockerswarm/
+  - /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/discovery.dockerswarm/
+  - /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/discovery.dockerswarm/
+canonical: https://grafana.com/docs/agent/latest/flow/reference/components/discovery.dockerswarm/
+title: discovery.dockerswarm
+---
+
+# discovery.dockerswarm
+
+`discovery.dockerswarm` allows you to retrieve scrape targets from [Docker Swarm](https://docs.docker.com/engine/swarm/key-concepts/).
+
+## Usage
+
+```river
+discovery.dockerswarm "LABEL" {
+  host = "unix:///var/run/docker.sock"
+  role = "services"
```

Comment:
For usage, it feels like we shouldn't provide example values, otherwise users may not understand that the role may sometimes not be `services` and that the host may be different (especially on Windows).

```suggestion
  host = "DOCKER_DAEMON_HOST"
  role = "SWARM_ROLE"
```